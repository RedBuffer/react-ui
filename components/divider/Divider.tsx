import * as React from 'react';
import classnames from 'classnames';

interface PropTypes {
  className?: string;

  style?: React.CSSProperties;

  children?: React.ReactNode;
}

const Divider = ({
  className,
  style,
  children,
  ...restProps
}: PropTypes): React.ReactElement => {
  return (
    <div
      className={classnames(
        'react-ui-divider',
        'border-gray-500',
        'border-t',
        'my-4',
        'p-0',
        'font-medium',
        'min-w-full',
        'w-full',
        children ? 'divider-with-text' : '',
        className,
      )}
      style={style}
      {...restProps}
    >
      {children ? (
        <span className={classnames('inline-block', 'px-2')}>{children}</span>
      ) : (
        <></>
      )}
    </div>
  );
};

export default Divider;
