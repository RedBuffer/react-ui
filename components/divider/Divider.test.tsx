import * as React from 'react';
import { shallow, mount } from 'enzyme';

import Divider from './Divider';

describe('Divider', () => {
  it('should render correctly', () => {
    const component = shallow(<Divider />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with text', () => {
    const component = shallow(<Divider>TEXT</Divider>);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with children', () => {
    const component = mount(
      <Divider>
        <button>CHILDREN</button>
      </Divider>,
    );

    expect(component).toMatchSnapshot();
  });
});
