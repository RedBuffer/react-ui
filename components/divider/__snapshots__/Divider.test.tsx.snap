// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render correctly 1`] = `
<div
  className="p-0 my-4 w-full border-t min-w-full font-medium border-gray-500 react-ui-divider"
/>
`;

exports[`Divider should render correctly with children 1`] = `
<Component>
  <div
    className="p-0 my-4 w-full border-t min-w-full font-medium border-gray-500 divider-with-text react-ui-divider"
  >
    <span
      className="px-2 inline-block"
    >
      <button>
        CHILDREN
      </button>
    </span>
  </div>
</Component>
`;

exports[`Divider should render correctly with text 1`] = `
<div
  className="p-0 my-4 w-full border-t min-w-full font-medium border-gray-500 divider-with-text react-ui-divider"
>
  <span
    className="px-2 inline-block"
  >
    TEXT
  </span>
</div>
`;
